### PR TITLE
[WFCORE-6346] Module java.base must open java.net package to all unnamed modules

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WFCORE-6346

### CODE REVIEW

**Comment by bstansberry** [comment]
@ropalka @fjuma This seems like something to fix in the 20.x branch as well, for this month's WF 28.0.1, as it seems like a regression in WF 28.0.0 via the upgrade that brought in https://issues.redhat.com/browse/WEJBHTTP-108.

**Comment by fjuma** [comment]
Good point, thanks @bstansberry.

**Comment by ropalka** [comment]
> @ropalka @fjuma This seems like something to fix in the 20.x branch as well, for this month's WF 28.0.1, as it seems like a regression in WF 28.0.0 via the upgrade that brought in https://issues.redhat.com/browse/WEJBHTTP-108.

Yes, I agree.

File: launcher/src/test/java/org/wildfly/core/launcher/CommandBuilderTest.java [hunk 1]
```diff
@@ -273,9 +273,11 @@ private void testJPMSArguments(final Collection<String> command, final int expec
         assertArgumentExists(command, "--add-exports=java.naming/com.sun.jndi.url.ldap=ALL-UNNAMED", expectedCount);
         assertArgumentExists(command, "--add-exports=java.naming/com.sun.jndi.url.ldaps=ALL-UNNAMED", expectedCount);
         assertArgumentExists(command, "--add-exports=jdk.naming.dns/com.sun.jndi.dns=ALL-UNNAMED", expectedCount);
+        assertArgumentExists(command, "--add-opens=java.base/java.lang=ALL-UNNAMED", expectedCount);
```
Comment (mchoma) [inline launcher/src/test/java/org/wildfly/core/launcher/CommandBuilderTest.java:276]: Seems as out of scope of this change.

Double checking if this is intentional.

File: core-feature-pack/common/src/main/resources/content/bin/common.bat [hunk 2]
```diff
@@ -50,10 +50,12 @@ goto :eof
       set "DEFAULT_MODULAR_JVM_OPTIONS=!DEFAULT_MODULAR_JVM_OPTIONS! --add-opens=java.base/java.lang=ALL-UNNAMED"
       rem Needed by the MicroProfile REST Client subsystem
       set "DEFAULT_MODULAR_JVM_OPTIONS=!DEFAULT_MODULAR_JVM_OPTIONS! --add-opens=java.base/java.lang.invoke=ALL-UNNAMED"
-      rem Needed by JBoss Marshalling
-      set "DEFAULT_MODULAR_JVM_OPTIONS=!DEFAULT_MODULAR_JVM_OPTIONS! --add-opens=java.base/java.io=ALL-UNNAMED"
       rem Needed for marshalling of proxies
       set "DEFAULT_MODULAR_JVM_OPTIONS=!DEFAULT_MODULAR_JVM_OPTIONS! --add-opens=java.base/java.lang.reflect=ALL-UNNAMED"
+      rem Needed by JBoss Marshalling
+      set "DEFAULT_MODULAR_JVM_OPTIONS=!DEFAULT_MODULAR_JVM_OPTIONS! --add-opens=java.base/java.io=ALL-UNNAMED"
```
[the remark this answers is not in the document]
Comment (yersan) [reply core-feature-pack/common/src/main/resources/content/bin/common.bat:56]: @mchoma This one should be fine, it was already there and down from L53 to keep it in the same order as it was declared in the other files

File: launcher/src/test/java/org/wildfly/core/launcher/CommandBuilderTest.java (hunk 1 above)
Comment (yersan) [reply launcher/src/test/java/org/wildfly/core/launcher/CommandBuilderTest.java:276]: I think this is fine; this test is testing that the expected JVM modular options are also used when launching jar files or the server for the tests

**Review by yersan — APPROVED** [review]
LGTM, so approving from my side

File: launcher/src/test/java/org/wildfly/core/launcher/CommandBuilderTest.java (hunk 1 above)
Comment (bstansberry) [reply launcher/src/test/java/org/wildfly/core/launcher/CommandBuilderTest.java:276]: Yeah, this was just a missing assert that Rio took the opportunity to correct.

**Review by jamezp — APPROVED** [review]

**Comment by fjuma** [comment]
@ropalka When you get a chance, please create a PR against the 20.x branch. Thanks!

**Comment by ropalka** [comment]
Here you go @fjuma : https://github.com/wildfly/wildfly-core/pull/5500

**Review by bstansberry — APPROVED** [review]
@ropalka This LGTM but if there isn't one already, please file a JIRA to update https://github.com/wildfly/wildfly/blob/main/pom.xml#L175-L207

**Comment by yersan** [comment]
Thanks, @ropalka and all involved in the review

**Pull request merged**